ocb3/src/lib.rs

```diff
@@ -580,11 +590,20 @@ pub(crate) fn ntz(n: usize) -> usize {
 }
 
 #[inline]
-pub(crate) fn split_into_two_blocks(two_blocks: &mut [u8]) -> [&mut Block; 2] {
-    const BLOCK_SIZE: usize = 16;
-    debug_assert_eq!(two_blocks.len(), BLOCK_SIZE * 2);
-    let (b0, b1) = two_blocks.split_at_mut(BLOCK_SIZE);
-    [b0.try_into().unwrap(), b1.try_into().unwrap()]
+pub(crate) fn split_into_two_blocks<'i, 'o>(
+    two_blocks: InOut<'i, 'o, DoubleBlock>,
+) -> [InOut<'i, 'o, Block>; 2] {
+    let (input, output) = two_blocks.into_raw();
+
+    let (bi0, bi1) = unsafe { input.as_ref() }
+        .unwrap()
+        .split_at(BlockSize::USIZE);
+    let (bi0, bi1): (&Block, &Block) = (bi0.try_into().unwrap(), bi1.try_into().unwrap());
+    let (bo0, bo1) = unsafe { output.as_mut() }
+        .unwrap()
+        .split_at_mut(BlockSize::USIZE);
```

Comment:
Those two unsafe blocks are less than ideal obviously, but I couldn't figure out how to make InOut split the array otherwise.

Reply:
something ala https://docs.rs/inout/0.2.0-rc.4/inout/struct.InOutBuf.html#method.split_at

Reply:
@newpavlov any opinion on this one?